app/controllers/host_controller.rb

```diff
@@ -136,6 +136,13 @@ def edit
     assert_privileges("host_edit")
     if session[:host_items].nil?
       @host = find_record_with_rbac(Host, params[:id])
+
+      if !@host.supports?(:update) # if host doesn't support editing
```

Comment:
I'm surprised rubocop didn't complain here, but prefer:

```suggestion
      unless @host.supports?(:update) # if host doesn't support editing
```